src/components/SearchInput/SearchInput.tsx

```diff
@@ -1,41 +1,55 @@
 import {
+  AriaAttributes,
   ChangeEvent,
-  ComponentProps,
   FocusEvent,
   MouseEvent,
   ReactNode,
   useEffect,
+  useRef,
   useState,
 } from "react"
 import styles from "./SearchInput.module.css"
 
-interface SearchInputProps extends ComponentProps<"input"> {
+interface SearchInputProps extends Pick<AriaAttributes, "aria-label"> {
+  name?: string
+  placeholder?: string
   searchIcon?: ReactNode
   clearIcon?: ReactNode
-  onDebounce?: (searchTerm: string) => void
   debounceDelay?: number
-  onSearchClick?: (searchTerm: string) => void
-  value?: string
+  initialValue?: string
+  onSearch?: (searchTerm: string) => void
+  showSearchButton?: boolean
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onFocus?: (e: FocusEvent<HTMLInputElement>) => void
 }
 
 export function SearchInput({
   name,
   placeholder,
   searchIcon,
   clearIcon,
-  onDebounce,
+  onSearch,
   debounceDelay = 300,
-  onSearchClick,
-  value = "",
-  ...inputProps
+  showSearchButton,
+  initialValue = "",
+  onBlur,
+  onFocus,
+  ...aria
 }: SearchInputProps) {
-  const [searchTerm, setSearchTerm] = useState(value)
-  const [hideCancelButton, setHideCancelButton] = useState(true)
+  const [searchTerm, setSearchTerm] = useState("")
+  const [cancelButtonHidden, setCancelButtonHidden] = useState(true)
+  const timer = useRef<NodeJS.Timeout>()
 
   useEffect(() => {
-    const timer = setTimeout(() => onDebounce?.(searchTerm), debounceDelay)
+    setSearchTerm(initialValue)
+  }, [initialValue])
```

Comment:
:pick: the prop name `initialValue` makes me think that if it changes there will be no effect inside the component's state. Maybe going back to `value` here could be okay? 😬

Reply:
I'm thinking of `initialValues` in [Formik](https://formik.org/docs/api/formik#initialvalues-values), which if changed do not update the current value of the form.

Reply:
Yeah, that makes sense. I was trying to get away from the common pattern of value and onChange, but your right, it isn't accurate since when that value changes, the state updates.